lib/wicked/wlan.pm

```diff
@@ -261,6 +262,45 @@ sub hostapd_can_wep {
     return $s !~ m/unknown configuration item 'wep_key0'/i;
 }
 
+sub hostapd_can_key_mgmt {
+    my ($self, $key_mgmt) = @_;
+    $self->write_cfg('/tmp/check_key_mgmt.conf', 'wpa_key_mgmt=' . $key_mgmt);
+    my $s = script_output('hostapd /tmp/check_key_mgmt.conf', proceed_on_failure => 1);
+    return $s !~ m/invalid key_mgmt/i;
+}
+
+sub wpa_supplicant_can_key_mgmt {
+    my ($self, $key_mgmt) = @_;
+    $self->write_cfg('/tmp/check_key_mgmt.conf', <<EOT);
+        network={
+            ssid=this-produce-a-ssid-parsing-error
+            key_mgmt=$key_mgmt
+        }
+EOT
+    my $s = script_output('wpa_supplicant -c/tmp/check_key_mgmt.conf -i ' . $self->sut_ifc, proceed_on_failure => 1);
+    return $s !~ m/invalid key_mgmt/i;
+}
+
+sub skip_by_supported_key_mgmt {
+    my ($self) = @_;
+    return 0 unless $self->need_key_mgmt;
+
+    if (!$self->hostapd_can_key_mgmt($self->need_key_mgmt)) {
+        record_info('SKIP', 'Skip test - hostapd does not support wpa_key_mgmt=' . $self->need_key_mgmt,
+            result => 'softfail');
+        $self->result('skip');
```

Comment:
ah I get it the overall module state is skip actually so all is fine . ignore comment above